### PR TITLE
Change_index_and_little_optimization_when_WaterUseType=0

### PR DESCRIPTION
**Pull request by Toubaal**

### CODE REVIEW

File: src/WaterUsePrepareRoutine.cpp [hunk 1]
```diff
@@ -108,6 +108,10 @@ void WaterUseCalcDaily(int waterUseType, NumericMatrix dailyUse, int year, int m
 					dailyUse(0,i) = GW_day[i];
 					dailyUse(1,i) = SW_day[i] + TF_day[i];
 		}
+				break;
+
+		//no water use is considered
+		case 0: dailyUse.fill(0); // Should not happen
```
Comment (JKupzig) [inline src/WaterUsePrepareRoutine.cpp:114]: Why do you have this case when you use an early return at the beginning of the function when the waterUseType is 0? 
Could you integrate an error if the given waterUseType is not 0,1 or 2? You can integrate it in the switch statement as default (https://www.w3schools.com/cpp/cpp_switch.asp).

Comment (Toubaal) [reply src/WaterUsePrepareRoutine.cpp]: I changed it :)
I also thought that only checking the settings so deep into the code was a bit weird so i added a settings checking at the beginning of runModel.

File: src/runModel.cpp [hunk 2]
```diff
@@ -28,6 +28,35 @@ using namespace Rcpp;
...
+	}
+	if (Settings[5] != 0  && Settings[5] != 1){
+		stop("splitttingFactor parameter should be 0 or 1");
+	}
+	if (Settings[6] != 0  && Settings[6] != 1){
+		stop("calculation LongWave parameter should be 0 or 1");
+	}
+	if (Settings[7] != 0  && Settings[7] != 1 && Settings[7] != 2 && Settings[7] != 3){
+		stop("useSystemVals should be 0, 1, 2 or 3");
+	}
```
Comment (JKupzig) [inline src/runModel.cpp:58]: This is really nice! 
What do you think about moving it inside defSettings? Then, one can immediately see which index refers to which setting.

Comment (Toubaal) [reply src/runModel.cpp]: I moved it, did you check that I check the right setting to be the right values ? You know the model better than me, maybe I did a mistake...

Comment (JKupzig) [reply src/runModel.cpp]: I've checked it - it looks fine. :-)

**Review by JKupzig — APPROVED** [review]

**Pull request merged**